### PR TITLE
Added Slash commands for quote commands

### PR DESCRIPTION
**Pull request by mikytron123**

### CODE REVIEW

File: NadekoBot.Core/Modules/Utility/QuoteSlashCommands.cs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                Quote quote;
+                await ctx.Interaction.DeferAsync().ConfigureAwait(false);
+                using (var uow = _db.GetDbContext())
+                {
+                    quote = await uow.Quotes.GetRandomQuoteByKeywordAsync(ctx.Guild.Id, keyword);
+                    //if (quote != null)
+                    //{
+                    //    quote.UseCount += 1;
+                    //    uow.Complete();
+                    //}
```
Comment (Gibstick) [inline NadekoBot.Core/Modules/Utility/QuoteSlashCommands.cs]: What's this commented code for?

File: NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs [hunk 2]
```diff
@@ -87,7 +87,7 @@ public static Task<string> ResponseWithContextAsync(this CustomReaction cr, IUse
 
         public static async Task<IUserMessage> Send(this CustomReaction cr, IUserMessage ctx, DiscordSocketClient client, bool sanitize)
         {
-            var channel = cr.DmResponse ? await ctx.Author.GetOrCreateDMChannelAsync().ConfigureAwait(false) : ctx.Channel;
+            var channel = cr.DmResponse ? await ctx.Author.CreateDMChannelAsync().ConfigureAwait(false) : ctx.Channel;
```
Comment (Gibstick) [inline NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs:90]: Is there a changelog that describes all of these changes in the discord.net library?

Comment (mikytron123) [reply NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs:90]: https://discordnet.dev/CHANGELOG.html

File: NadekoBot.Core/Modules/Utility/QuoteSlashCommands.cs (hunk 1 above)
Comment (mikytron123) [reply NadekoBot.Core/Modules/Utility/QuoteSlashCommands.cs]: it was commented in QuoteCommands.cs and i just copied that. I can delete it if you want

File: NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs (hunk 2 above)
Comment (Gibstick) [reply NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs:90]: I don't see anything specific about all these big changes. You're sure they're absolutely necessary? They're not really covered in the migration guide either https://discordnet.dev/guides/v2_v3_guide/v2_to_v3_guide.html

Comment (mikytron123) [reply NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs:90]: Yes it is necessary, that method was removed in version 3.1

Comment (mikytron123) [reply NadekoBot.Core/Modules/CustomReactions/Extensions/Extensions.cs:90]: > I don't see anything specific about all these big changes. You're sure they're absolutely necessary? They're not really covered in the migration guide either https://discordnet.dev/guides/v2_v3_guide/v2_to_v3_guide.html

https://discordnet.dev/api/Discord.IUser.html No getorcreatedmchannelasync method here

**Review by Gibstick — APPROVED** [review]

**Pull request merged**